File: Frontend/src/modules/gallery/services/galleryService.js
import api from '@/services/api';

const galleryService = {
  getAll: async () => {
    const response = await api.get('/galleries');
    return response.data;
  },
  getById: async (id) => {
    const response = await api.get(`/galleries/${id}`);
    return response.data;
  },
  getMedia: async (id) => {
    const response = await api.get(`/galleries/${id}/media`);
    return response.data;
  },
  create: async (data) => {
    const response = await api.post('/galleries', data);
    return response.data;
  },
  uploadMedia: async (id, file) => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await api.post(`/galleries/${id}/upload`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' }
    });
    return response.data;
  },
  deleteMedia: async (mediaId) => {
    const response = await api.delete(`/media/${mediaId}`);
    return response.data;
  }
};

export default galleryService;

File: Backend/src/main/java/com/family/service/GalleryService.java
package com.family.service;

import com.family.entity.Gallery;
import com.family.entity.Media;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.UUID;

public interface GalleryService {
    Gallery getById(UUID id);
    List<Gallery> getAll();
    Gallery create(Gallery gallery);
    void delete(UUID id);
    List<Media> getMediaByGalleryId(UUID galleryId);
    Media uploadMedia(UUID galleryId, MultipartFile file);
    void deleteMedia(UUID mediaId);
}
